tests/integration/v2/app.go

```diff
@@ -116,12 +119,30 @@ func DefaultStartUpConfig(t *testing.T) StartupConfig {
 	}
 }
 
+// RunMsgConfig defines the run message configuration.
+type RunMsgConfig struct {
+	Commit bool
+}
+
+// Option is a function that can be used to configure the integration app.
+type Option func(*RunMsgConfig)
+
+// WithAutomaticCommit enables automatic commit.
+// This means that the integration app will automatically commit the state after each msg.
+func WithAutomaticCommit() Option {
+	return func(cfg *RunMsgConfig) {
+		cfg.Commit = true
+	}
+}
+
 // NewApp initializes a new runtime.App. A Nop logger is set in runtime.App.
 // appConfig defines the application configuration (f.e. app_config.go).
 // extraOutputs defines the extra outputs to be assigned by the dependency injector (depinject).
 func NewApp(
 	appConfig depinject.Config,
 	startupConfig StartupConfig,
+	branchService corebranch.Service,
```

Comment:
Let's make it an option struct or include this in the `StartupConfig` struct.
It is not nice to pass nil around.

Let's precise the reason why you would pass those custom services in the constructor doc